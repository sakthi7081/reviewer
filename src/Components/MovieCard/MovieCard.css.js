import styled from 'styled-components';
import {CardMedia} from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import Rating from '@material-ui/lab/Rating';
import {Typography} from '@material-ui/core';

export const StyledCardMedia = styled(CardMedia)`
        height : 350px;        
        &&&{
                background-size : 100% 100%;
        }
`

export const StyledSynopsis = styled(Typography)`
        height : 100px;        
        overflow: hidden;
        display: -webkit-box;
        -webkit-line-clamp: 5;
        -webkit-box-orient: vertical;
`

export const StyledRating = withStyles({
        iconFilled: {
          color: '#ff3d47',
        },
        iconHover: {
          color: '#ff3d47',
        },
        
      })(Rating);
