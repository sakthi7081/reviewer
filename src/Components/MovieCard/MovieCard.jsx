import React from 'react';
import {StyledCardMedia,StyledRating,StyledSynopsis} from './MovieCard.css';
import FavoriteIcon from '@material-ui/icons/Favorite';
import {Card,CardActionArea,CardContent,Typography} from '@material-ui/core';
import Img from '../../Utilities/blank_poster.jpg';
import Skeleton from '@material-ui/lab/Skeleton';

export const  MovieSkeleton = () => {        
    return (        
        <Card >
            <CardActionArea>
                <Skeleton animation="wave" height={350} variant="rect" />
                <CardContent>
                    <Skeleton animation="wave" height={40} width={'40%'} />   
                    <div style={{display : 'flex'}}>
                        <Skeleton animation="wave" height={10} width={'25%'} />
                        {` \u25CF`}
                        <Skeleton animation="wave" height={10} width={'25%'} />                          
                    </div>                                
                <StyledRating                    
                    value={0}
                    readOnly                     
                    precision={0.5}
                    disabled
                    icon={<FavoriteIcon fontSize="inherit" />}
                    />
                <StyledSynopsis variant="body2" component="p">
                    <Skeleton animation="wave" height={20} width={'100%'} />                
                    <Skeleton animation="wave" height={20} width={'100%'} />                
                    <Skeleton animation="wave" height={20} width={'100%'} />                
                    <Skeleton animation="wave" height={20} width={'100%'} /> 
                    <Skeleton animation="wave" height={20} width={'50%'} />                
                </StyledSynopsis>
                </CardContent>
            </CardActionArea>            
        </Card>        
    );
}


const MovieCard = ({movie}) => {    
    const rating = movie.ratings.map(value=> value.rating).reduce(((a,b)=> a+b),0);    
    return (        
        <Card >
            <CardActionArea>
                <StyledCardMedia                
                image={movie.img === '' ? Img : movie.img}   
                src={movie.img_name ? movie.img_name : ''}
                />
                <CardContent style={{height: '180px'}}>
                <Typography variant="h5" component="h1" >
                    {movie.name}
                </Typography>
                <Typography variant="caption" display="block" gutterBottom>
                    {`${(movie.date).toLocaleDateString()} \u25CF ${movie.language}`}
                </Typography>               
                <StyledRating                    
                    value={rating}
                    readOnly                     
                    precision={0.5}
                    icon={<FavoriteIcon fontSize="inherit" />}
                    />
                <StyledSynopsis variant="body2" component="p">
                    {movie.description}
                </StyledSynopsis>
                </CardContent>
            </CardActionArea>            
        </Card>        
    );
}


export default MovieCard;