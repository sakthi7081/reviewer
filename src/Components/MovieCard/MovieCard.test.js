import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from "enzyme-to-json";
import MovieCard from './MovieCard';

const movie = {
    _id :"5e9411ab7b32077f7c7daa13",
    name :"The eagle eye",
    description :"Eagle eye",
    language :"English",
    date :new Date("2020-04-13T07:14:23.695Z"),
    img :"",
    img_name :"",
    created_on :new Date("2020-04-13T07:15:55.874Z"),
    created_by :{
       id:"5e9411677b32077f7c7daa12",
       username:"Saravanan"
    },
    ratings:[
       {
          comment:"Everybody don't miss that ðŸ˜‰",
          rating:5,
          id:"5e9411677b32077f7c7daa12",
          username:"Saravanan"
       }
    ]
 };

const MovieWrapper = shallow(<MovieCard movie={movie}/>);

describe("MovieCard", () => {
  it("should render Movie Card", () => {
    expect(shallowToJson(MovieWrapper)).toMatchSnapshot();
  });
});