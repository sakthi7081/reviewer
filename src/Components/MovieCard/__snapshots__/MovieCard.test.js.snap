// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should render Movie Card 1`] = `
<WithStyles(ForwardRef(Card))>
  <WithStyles(ForwardRef(CardActionArea))>
    <Styled(WithStyles(ForwardRef(CardMedia)))
      image="blank_poster.jpg"
      src=""
    />
    <WithStyles(ForwardRef(CardContent))
      style={
        Object {
          "height": "180px",
        }
      }
    >
      <WithStyles(ForwardRef(Typography))
        component="h1"
        variant="h5"
      >
        The eagle eye
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        display="block"
        gutterBottom={true}
        variant="caption"
      >
        4/13/2020 ‚óè English
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(WithStyles(ForwardRef(Rating)))
        icon={
          <UNDEFINED
            fontSize="inherit"
          />
        }
        precision={0.5}
        readOnly={true}
        value={5}
      />
      <Styled(WithStyles(ForwardRef(Typography)))
        component="p"
        variant="body2"
      >
        Eagle eye
      </Styled(WithStyles(ForwardRef(Typography)))>
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(CardActionArea))>
</WithStyles(ForwardRef(Card))>
`;
